src/Models/Model.php

```diff
@@ -190,29 +232,31 @@ public static function getPropertyDefinition( string $key ): ModelPropertyDefini
 	 * @since 2.0.0
 	 *
 	 * @return array<string,ModelPropertyDefinition>
+	 *
+	 * @throws InvalidArgumentException If the property key is not a string.
 	 */
 	public static function getPropertyDefinitions(): array {
-		static $definition = null;
-
-		if ( $definition === null ) {
-			$definitions = array_merge( static::$properties, static::properties() );
+		if ( isset( static::$cached_definitions[ static::class ] ) ) {
+			return static::$cached_definitions[ static::class ];
+		}
 
-			foreach ( $definitions as $key => $definition ) {
-				if ( ! is_string( $key ) ) {
-					throw new \InvalidArgumentException( 'Property key must be a string.' );
-				}
+		$definitions = array_merge( static::$properties, static::properties() );
 
-				if ( ! $definition instanceof ModelPropertyDefinition ) {
-					$definition = ModelPropertyDefinition::fromShorthand( $definition );
-				}
+		foreach ( $definitions as $key => $definition ) {
+			if ( ! is_string( $key ) ) {
+				throw new InvalidArgumentException( 'Property key must be a string.' );
+			}
 
-				$definitions[ $key ] = $definition->lock();
+			if ( ! $definition instanceof ModelPropertyDefinition ) {
+				$definition = ModelPropertyDefinition::fromShorthand( $definition );
 			}
 
-			$definition = $definitions;
+			$definitions[ $key ] = $definition->lock();
```

Comment:
I feel like the branch this is targeted to merge into isn't correct, @dpanta94? I believe I remember doing this a while back, but maybe I'm wrong? This doesn't have anything to do with notices and type hints.